### PR TITLE
DHFPROD-4570: User admin cannot write artifacts to ML DB

### PR DESCRIPTION
**Pull request by hao1st**
https://project.marklogic.com/jira/browse/DHFPROD-4570

### CODE REVIEW

**Comment by bsrikan** [comment]
Didnt get a chance to review yet but the commit message is confusing me. With this PR we are allowing an admin to write artifacts to ML right?

**Comment by hao1st** [comment]
@bsrikan Yes, and the ticket is a bug to indicate that admin cannot write artifacts now. The PR is fixing the problem.

File: one-ui/src/test/java/com/marklogic/hub/curation/controllers/EnvironmentControllerTest.java [hunk 1]
```diff
@@ -248,6 +253,29 @@ public void doAppend(ILoggingEvent event) throws LogbackException {
...
+
+    @Test
+    public void testAdminAuthoritiesForArtifacts()  {
+        EnvironmentInfo environmentInfo = new EnvironmentInfo("localhost", "DIGEST", 8000,"DIGEST", 8002,"DIGEST", 8010, "DIGEST", 8011);
+        hubConfigSession.setCredentials(environmentInfo, testHelper.adminUserName, testHelper.adminPassword);
+        ArtifactManager mgr = testHelper.getArtifactManager();
+        List<ArtifactTypeInfo> listTypeInfo = mgr.getArtifactTypeInfoList();
+        for (ArtifactTypeInfo typeInfo : listTypeInfo) {
+            assertTrue(typeInfo.getUserCanUpdate());
+            assertFalse(typeInfo.getUserCanRead());
```
Comment (rjrudin) [inline one-ui/src/test/java/com/marklogic/hub/curation/controllers/EnvironmentControllerTest.java]: Can you add an assertion message to explain why an admin user can update, but not read? That of course is surprising behavior - normally a user can read but not update (or can do both). According to the JIRA ticket, it's to allow an admin to write artifacts during deployment - I think it's important that an assertion message captures that.

**Comment by bsrikan** [comment]
There is a bug today in that we dont save the <loadData/mappings>.settings.json in the filesystem. I will file a separate bug for that.
Nevertheless this PR is trying to write all the user artifacts to ML. Hence loadData.settings.json/ mapping.settings.json/matching.settings.json/merging.settings.json should be considered for being written into ML as well as part of deploy.

**Comment by hao1st** [comment]
@bsrikan Can you let me know the bug number? I am not sure I am getting at the scenario you mentioned.

**Comment by bsrikan** [comment]
Discussed with Hao after stand up. Filed a bug for settings.json to be saved in filesystem https://project.marklogic.com/jira/browse/DHFPROD-4608 which can be addressed outside of this PR as well.

**Comment by hao1st** [comment]
@bsrikan Please check.

**Pull request merged**